src/server/ua_services_attribute.c

```diff
@@ -1097,16 +1097,29 @@ writeValueAttribute(UA_Server *server, UA_Session *session,
         else
             retval = writeValueAttributeWithRange(node, &adjustedValue, rangeptr);
 
+#ifdef UA_ENABLE_HISTORIZING
+        if(retval == UA_STATUSCODE_GOOD && server->config.historyDataService.setValue)
```

Comment:
Check if the historizing attribute is true.
Only call `setValue` if that is the case.

What is the interaction between calling `setValue` here and a variable that is monitored with a local MonitoredItem for the HistoricalData? I presume for the latter case, the `historizing` attribute would be true as well?

Reply:
It should be possible to add values to the database even when the historizing attribute is not set.
But it is a good idea to provide the historizing attribute to the setValue call.
This is done.